Library/Homebrew/extend/ENV/super.rb

```diff
@@ -68,6 +68,10 @@ def setup_build_environment(formula = nil)
     self["HOMEBREW_INCLUDE_PATHS"] = determine_include_paths
     self["HOMEBREW_LIBRARY_PATHS"] = determine_library_paths
     self["HOMEBREW_DEPENDENCIES"] = determine_dependencies
+    unless formula.nil?
+      self["HOMEBREW_FORMULA"] = formula.name
+      self["HOMEBREW_FORMULA_VERSION"] = formula.version
```

Comment:
I'm not actually sure about this: there are lots of `brew` self-calls, and it may be appropriate for it to be inherited from the parent process in some of them if there's a self-call running inside an operation for a specific formula. I'll poke around and see if I can come up with an example or counter-example.